model/agent.py

```diff
@@ -0,0 +1,300 @@
+from model.library.enums import AgentType
+from model.library.functions import clamp, cognative_temperature_curve
+from model.library.objects import Context, SampleData
+
+class Person:
+    """Agent Parent Class"""
+    agent_type = ""
+    # TODO: Add system stability affect attention level
+    # If the system is unstable attention should increase, if system is stable attention will decrease
+    # We know what the status of the system is use this to callibrate the system
+    # begining they will be focused, not exausted yet, 4pm things go wrong and they are tired
+
+    # TODO: Analygous attentional properties will attach to data analyst
+    # Hot vs cold cognition, hot rappid makes more mistakes, cold slower more accurate
+    # Eventually check when worrying check all the time, when not check once in a while
+    # TODO: Attention/ exaustion/ focus controlls for ever person
+    # TODO _: Add Attention
+    # TODO: begining they will be focused, not exausted yet, 4pm things go wrong and they are tired
+    cognative_temperature:float = None
+    cogtemp_curve:float = None
+    # 1.0 / 0.002 = 8.3, total level of energy / minutes = 8.3 total hours of energy
+    energy_degradation:float = 0.002
+
+    attention_meter:float = None
+
+
+    previous_check:timedelta = None
+    noticing_delay:float = 1.0  # 100 ms
+    decision_delay:float = 1.0  # 100 ms -- FFF incorporate differential switch time
+    functional_acuity:float = 0.01
+
+    def __init__(self, agent_type):
+        self.agent_type = agent_type
+        self.cognative_temperature = 1.0
+        self.cogtemp_curve = 0.0
+        self.attention_meter = 1.0
+        self.previous_check = timedelta(0)
+
+    def get_energy(self):
+        """get the level of energy"""
+        return self.cognative_temperature
+
+    def get_attention(self):
+        """get the level of attention"""
+        return self.cogtemp_curve
+
+    def update(self, context:Context):
+        """Update attention and focus"""
+        # TODO _: energy degredation on a curve
+        # TODO: coupled also on a curve
+        delta:timedelta = context.current_time - self.previous_check
+        if delta >= timedelta(minutes=1):
+            self.cogtemp_curve = clamp(self.cogtemp_curve + self.energy_degradation, 0.0, 1.0)
+            self.cognative_temperature = clamp(cognative_temperature_curve(self.cogtemp_curve), 0.01, 1.0)
+            self.previous_check = context.current_time
+        self.noticing_delay = 1 + (1 - self.cognative_temperature)
+        self.decision_delay = 1 + (1 - self.cognative_temperature)
+
+class DataAnalyst(Person):
+    """Retrives data from the instrument"""
+    projected_intercept:float = None
+    def __init__(self):
+        super().__init__(AgentType.DA)
+        self.last_sample_with_enough_data = None
+
+    def check_if_enough_data_to_analyse(self, context:Context) -> bool:
+        """Check if there is enough data to start analysing, right now this is a constant"""
```

Comment:
This is called every cycle, right now the prediction logic is commented out so that whoever the sample count gets above constant it will start checking error and mean